Documentation/development-environment.md

```diff
@@ -0,0 +1,125 @@
+download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 15.39s
+kubernetes/master : Master | wait for the apiserver to be running ------ 12.44s
+download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 11.83s
+download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 11.66s
+kubernetes/node : install | Copy kubelet from hyperkube container ------ 11.44s
+download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 11.41s
+download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 11.00s
+docker : Docker | pause while Docker restarts -------------------------- 10.22s
+kubernetes/secrets : Check certs | check if a cert already exists on node --- 6.05s
+kubernetes-apps/network_plugin/flannel : Flannel | Wait for flannel subnet.env file presence --- 5.33s
+kubernetes/master : Master | wait for kube-scheduler -------------------- 5.30s
+kubernetes/master : Copy kubectl from hyperkube container --------------- 4.77s
+[leseb@tarox kubespray]$
+[leseb@tarox kubespray]$
+[leseb@tarox kubespray]$ vagrant ssh k8s-01
+Last login: Fri Jan 12 09:22:18 2018 from 192.168.121.1
+
+[vagrant@k8s-01 ~]$ kubectl get nodes
+NAME      STATUS    ROLES         AGE       VERSION
+k8s-01    Ready     master,node   2m        v1.9.0+coreos.0
+k8s-02    Ready     node          2m        v1.9.0+coreos.0
+k8s-03    Ready     node          2m        v1.9.0+coreos.0
+k8s-04    Ready     node          2m        v1.9.0+coreos.0
+k8s-05    Ready     node          2m        v1.9.0+coreos.0
+k8s-06    Ready     node          2m        v1.9.0+coreos.0
+k8s-07    Ready     node          2m        v1.9.0+coreos.0
+k8s-08    Ready     node          2m        v1.9.0+coreos.0
+k8s-09    Ready     node          2m        v1.9.0+coreos.0
+```
+
+## Development workflow
+
+Install git on k8s-01:
+
+```
+[vagrant@k8s-01 ~]$ sudo yum install git -y
+```
+
+Now, please refer to [https://rook.io/docs/rook/master/development-flow.html](https://rook.io/docs/rook/master/development-flow.html) to setup your development environment (go, git etc).
+
+At this stage, Rook should be cloned on k8s-01.
+
+From your Rook repository (should be $GOPATH/src/github.com/rook) location execute `bash tests/scripts/multi-node/build-rook.sh`.
+During its execution `build-rook.sh` also purge your Rook pod so Rook container images can be build.
+Now that Rook pods are gone with have to purge the Ceph data dir and disks.
+For this, place `tests/scripts/multi-node/purge-ceph.sh` in your Kubespray git root directory. and execute it after `build-rook.sh`.
+
+Deploy again, virtual machines (k8s-0X) will pull the new container image and run your new Rook code.
```

Comment:
what exactly is meant by "deploy again" here, does that mean the normal `kubectl create -f rook-operator.yaml` etc.?

Reply:
Yes, let me clarify that.